jax/_src/lax/lax.py

```diff
@@ -2001,8 +2001,7 @@ def _abs_jvp_rule(g, ans, x):
 pow_p = standard_naryop([_float | _complex, _float | _complex], 'pow')
 
 def _pow_jvp_lhs(g, ans, x, y):
-  jac = mul(y, pow(x, select(eq(y, _zeros(y)), _ones(y), sub(y, _ones(y)))))
-  return mul(g, jac)
+  return mul(g, mul(y, pow(x, sub(y, _ones(y)))))
```

Comment:
It's unclear to me why this rule ever included a select statement – @mattjj, @froystig, do you have any recollection why it was originally implemented this way?

Reply:
I found the PR that implemented the orginal logic: https://github.com/google/jax/pull/383

Reply:
Some git blame sleuthing by @mattjj found the origin of this select statement: https://github.com/HIPS/autograd/issues/116

Reply:
Whoever wrote that line is an idiot!